app/models/katello/content_view.rb

```diff
@@ -770,43 +770,30 @@ def related_composite_cvs
       content_views
     end
 
-    def audited_cv_repositories_since_last_publish
-      audited_repositories = self.audits.filter do |a|
+    def audited_cv_repositories_since_last_publish?
+      self.audits.where("created_at > ?", latest_version_object.created_at).any? do |a|
```

Comment:
this would run another SQL query. Can this be prevented?

Reply:
The `self.audit` should be fast since the foreign key fields are indexed. The alternative would be to run a `Audit.where..` query which could merge those.

Rails also seems to be optimizing the self.audits relation and where clause into one query.
```
[2] pry(main)> cv.audited_cv_repositories_since_last_publish?
2024-02-29T19:57:37 [D|sql|]   Katello::ContentViewVersion Load (0.6ms)  SELECT "katello_content_view_versions".* FROM "katello_content_view_versions" WHERE "katello_content_view_versions"."content_view_id" = $1 ORDER BY major DESC, minor DESC LIMIT $2  [["content_view_id", 2], ["LIMIT", 1]]
2024-02-29T19:57:37 [D|sql|]   Audited::Audit Load (0.8ms)  SELECT "audits".* FROM "audits" WHERE "audits"."auditable_id" = $1 AND "audits"."auditable_type" = $2 AND (created_at > '2024-02-29 17:12:01.448064') ORDER BY "audits"."version" ASC  [["auditable_id", 2], ["auditable_type", "Katello::ContentView"]]
=> false

```

Reply:
Well, better would be, that all this stuff is not loaded in separate queries and at least not n+1. Use eager loading and use already loaded data if possible.

Reply:
Aah..I see..I see your PR for cv components along the same lines. Le me try that for audits..
https://github.com/Katello/katello/pull/10911